istio-telemetry/kiali-operator/templates/kiali.yaml

```diff
@@ -0,0 +1,425 @@
+apiVersion: kiali.io/v1alpha1
+kind: Kiali
+metadata:
+  name: kiali
+  namespace: {{ .Release.Namespace }}
+  labels:
+    release: {{ .Release.Name }}
+annotations:
+  ansible.operator-sdk/reconcile-period: "0s"
+spec:
+  ###################################################################
+  # kiali_cr.yaml
+  #
+  # This is a fully documented Kiali custom resource yaml file.
+  # It can be used to tell the Kiali Operator to install Kiali.
+  #
+  # This is actually an empty Kiali CR, however, it provides
+  # documentation on all available settings.
+  # In each documented section, you will see a "---" marker;
+  # below that marker you will see the names of the settings along
+  # with their default values. If the setting is not defined by
+  # default, its name will be prefixed with "#".
+  ###################################################################
+
+  ##########
+  # Tag used to identify a particular instance/installation of the Kiali server.
+  #    ---
+  #  installation_tag: ""
+
+  ##########
+  # The namespace where Istio is installed. If left empty, it is assumed to be the
+  # same namespace as where Kiali is installed (i.e. deployment.namespace).
+  #    ---
+  #  istio_namespace: ""
+  istio_namespace: {{ .Release.Namespace }}
```

Comment:
what is "istio_namespace" in kiali? Pilot namespace? Kiali may be deployed in its own namespace so I would guess `.Release.Namespace` may not be correct

Reply:
Good catching. I have browsed the kiali source code. the `istio_namespace` is used in:
1. discover Tracing Service if no defined in `jaegerURL`
2. discover prometheus service if no defined in `prometheusAddr`
3. graph related logic are using. e.g.: https://github.com/kiali/kiali/blob/82a3496fd62d7ae0756403e84676ee040c39acab/graph/telemetry/istio/appender/security_policy.go#L70
4. handler related logic. e.g.: https://github.com/kiali/kiali/blob/master/handlers/threescale.go

That is not related with kiali-operator. I think the kiali should have some issues for new installer which is recommended to install istio components in different namespaces. @jmazzitelli any comments? Thanks.

Reply:
This may require some thinking/changing on Kiali's part. Today, there is a single "istio_namespace" setting in the Kiali configuration - it is assumed Kiali components such as jaeger and prometheus by default are located here. See: https://github.com/kiali/kiali/blob/v1.1.0/operator/deploy/kiali/kiali_cr.yaml#L28-L32

Though there are settings you can use to override them - like:

Grafana: https://github.com/kiali/kiali/blob/v1.1.0/operator/deploy/kiali/kiali_cr.yaml#L178-L180
Prometheus: https://github.com/kiali/kiali/blob/v1.1.0/operator/deploy/kiali/kiali_cr.yaml#L223-L224
Jaeger: https://github.com/kiali/kiali/blob/v1.1.0/operator/deploy/kiali/kiali_cr.yaml#L255-L257 

That [graph related logic](https://github.com/kiali/kiali/blob/82a3496fd62d7ae0756403e84676ee040c39acab/graph/telemetry/istio/appender/security_policy.go#L70) is to filter out the graph all istio-related nodes so you don't get a very noisy graph of all the istio traffic going to the istio telemetry pods.

I'm not familiar with the threescale usage of this istio_namespace setting.

Reply:
I have logged an issue in kiali side to trace how to achieve it - https://github.com/kiali/kiali/issues/1315.  Thanks.

Reply:
For 1.3 we may keep everything in a single namespace. But we really want to move at least prometheus/grafana to their own operators and separate ns.

Reply:
> But we really want to move at least prometheus/grafana to their own operators and separate ns.

And that is already supported today by Kiali.. The operator can be told about different locations for Prometheus and Grafana and Jaeger:

Grafana: https://github.com/kiali/kiali/blob/v1.2/operator/deploy/kiali/kiali_cr.yaml#L196
Prometheus: https://github.com/kiali/kiali/blob/v1.2/operator/deploy/kiali/kiali_cr.yaml#L239-L240
Jaeger: https://github.com/kiali/kiali/blob/v1.2/operator/deploy/kiali/kiali_cr.yaml#L273-L275